src/main/java/com/github/maven_nar/cpptasks/compiler/AbstractProcessor.java

```diff
@@ -67,7 +67,16 @@ protected static String getIdentifier(final String[] command, final String fallb
 
   protected AbstractProcessor(final String[] sourceExtensions, final String[] headerExtensions) {
     this.sourceExtensions = sourceExtensions.clone();
-    this.headerExtensions = headerExtensions.clone();
+      if(headerExtensions != null)
```

Comment:
Rather than changing the contract that all other compilers match, perhaps follow existing pattern and pass 
`new String[] {}`